prover/src/universal_batch_verifier/compute_submission_id.rs

```diff
@@ -0,0 +1,54 @@
+use circuits::{
+    batch_verify::universal::native::{
+        compute_circuit_id, json::load_app_vk_proof_and_inputs_batch,
+    },
+    keccak::utils,
+};
+use clap::Parser;
+use core::iter;
+
+#[derive(Parser, Debug)]
+pub struct ComputeSubmissionIdParams {
+    /// JSON file containing a batch of app_vk, proof, public input triples
+    /// to be verified by the BatchVerifier circuit.
+    #[arg(short = 'b', long, value_name = "app-vk-proof-batch-file")]
+    pub(crate) app_vk_proof_batch: String,
+
+    /// Num proof ids
+    #[arg(short = 'n', long, value_name = "num_proof_ids")]
+    pub(crate) num_proof_ids: Option<usize>,
+
+    /// Submission size
+    #[arg(short = 's', long, value_name = "submission-size")]
+    pub(crate) submission_size: Option<usize>,
+}
+
+pub fn compute_submission_id(params: ComputeSubmissionIdParams) {
+    let ubv_inputs =
+        load_app_vk_proof_and_inputs_batch(&params.app_vk_proof_batch);
+    let batch_len = ubv_inputs.0.len();
+    let num_proof_ids = params.num_proof_ids.unwrap_or(batch_len);
+    assert!(
+        num_proof_ids <= batch_len,
+        "Batch length must be greater than or equal to the number of proof ids"
+    );
+    let next_power_of_two = || {
+        if num_proof_ids.is_power_of_two() {
+            num_proof_ids
+        } else {
+            1usize << (64 - num_proof_ids.leading_zeros())
```

Comment:
I'm a bit nervous about using `leading_zeros` on a `usize` here because it probably behaves differently in 32-bit architecture than 64-bit. The docs for this function suggest trying `ilog2` instead, that might be the way to do it. 

Or maybe cast to `u64` and use https://doc.rust-lang.org/std/primitive.u64.html#method.next_power_of_two

Reply:
Good catch